### PR TITLE
rename updates

### PR DESCRIPTION
**Pull request by tom-wa**
like this ?
get/case will be applied initially to the keyset, then the normal transformations are done.
on set the keyname will be either:
 restored and converted to lower or uppercase depending on set/case
of
if set/case == keyname: not restored and the transformed name kept.

### CODE REVIEW

File: src/plugins/rename/rename.c [hunk 1]
```diff
@@ -315,38 +333,84 @@ int elektraRenameSet (Plugin * handle, KeySet * returned, Key * parentKey)
...
 	ksDel (iterateKs);
 
+	ksRewind (returned);
+	elektraFree (parentKeyName);
 	return 1; /* success */
 }
 
 Plugin * ELEKTRA_PLUGIN_EXPORT (rename)
 {
-	// clang-format off
```
Comment (markus2330) [inline src/plugins/rename/rename.c]: I think its nicer when you leave this customized formatted, don't you agree?

**Comment by markus2330** [comment]
Thank you! Now only the profile plugin and we should be done with new features for this release!

Can you add test cases for the new features?

There are memory problems reported by build server.

I do not have time for a closer look today, sorry!

File: src/plugins/rename/rename.c (hunk 1 above)
Comment (tom-wa) [inline src/plugins/rename/rename.c]: i honestly have no idea why that line always disappears...

**Comment by tom-wa** [comment]
@markus2330 done

File: src/plugins/rename/rename.c [hunk 2]
```diff
@@ -1,9 +1,9 @@
 /**
- * @file
+ * \file
```
Comment (markus2330) [inline src/plugins/rename/rename.c]: @ is the correct one, see doc/CODING.md

**Comment by markus2330** [comment]
What does the `keyname` do? Please add info in the README.

**Comment by markus2330** [comment]
Any progress here? The issues should only be minor?

**Comment by tom-wa** [comment]
@markus2330 done

File: src/plugins/rename/rename.c [hunk 3]
```diff
@@ -203,23 +228,16 @@ static Key * restoreKeyName (Key * key, const Key * parentKey, const Key * confi
 	{
 		if (strcmp (keyString (origNameKey), keyName (key)))
 		{
-			int hasSync = keyNeedSync (key); // test_bit(key->flags, KEY_FLAG_SYNC);
```
Comment (markus2330) [inline src/plugins/rename/rename.c]: Why did you remove these lines which restore the sync bit? They are needed for #404. We should always put comments above lines that fix a bug so that everyone know what they are for.

File: src/plugins/rename/rename.c [hunk 4]
```diff
@@ -1,9 +1,9 @@
 /**
  * @file
  *
- * @brief A plugin that converts keys to metakeys and vice versa
+ * \brief A plugin that converts keys to metakeys and vice versa
```
Comment (markus2330) [inline src/plugins/rename/rename.c]: please change, it should be `@`

File: src/plugins/rename/README.md [hunk 5]
```diff
@@ -16,6 +16,34 @@ with the required names or case.
...
+* basic
+* advanced
+
+
+## BASIC TRANSFORMATIONS ##
+
+are applied before and after the advanced transformations.
+
+### GET ###
+`get/case`
```
Comment (markus2330) [inline src/plugins/rename/README.md]: one sentence here would be good.

File: src/plugins/rename/README.md [hunk 6]
```diff
@@ -16,6 +16,34 @@ with the required names or case.
...
+* toupper
+* tolower
+* keyname
+* unchanged   // this is the default value if no configuration is given
+
+`toupper` or `tolower` tells the rename plugin to convert the whole keyname below below the parentKey to lower or uppercase.
+`unchanged` returnes the key to it's original name
+`keyname` tells the plugin to keep the name of the advanced transformation
+
+## ADVANCED TRANSFORMATIONS ##
```
Comment (markus2330) [inline src/plugins/rename/README.md]: Correct level of title? (`#` instead of `##`?)

File: src/plugins/list/testmod_list.c [hunk 7]
```diff
@@ -31,7 +31,6 @@ static void doTest ()
 			       keyNew ("user/plugins/#1/placements/set", KEY_VALUE, "presetstorage", KEY_END),
 			       keyNew ("user/plugins/#1/placements/get", KEY_VALUE, "pregetstorage", KEY_END), KS_END);
 	Key * parentKey = keyNew ("user/tests/list", KEY_END);
-	ksAppendKey (ks, parentKey);
```
Comment (markus2330) [inline src/plugins/list/testmod_list.c]: should not be here?

File: src/plugins/rename/rename.c [hunk 8]
```diff
@@ -207,11 +234,6 @@ static Key * restoreKeyName (Key * key, const Key * parentKey, const Key * confi
 			Key * result = keyDup (key);
 			keySetName (result, keyString (origNameKey));
 			keySetMeta (result, ELEKTRA_ORIGINAL_NAME_META, 0);
-
-			if (!hasSync)
```
Comment (markus2330) [inline src/plugins/rename/rename.c]: Still removed. Could you also please add a comment that it is here because of #404. (At all places)

File: src/plugins/rename/README.md [hunk 9]
```diff
@@ -8,14 +8,51 @@
...
+* toupper
+* tolower
+* keyname
+* unchanged   // this is the default value if no configuration is given
+
+`toupper` or `tolower` tells the rename plugin to convert the whole keyname below below the parentKey to lower or uppercase.
+
+`unchanged` returnes the key to it's original name
+
+`keyname` tells the plugin to keep the name of the advanced transformation
```
Comment (markus2330) [inline src/plugins/rename/README.md]: one example would be great!

File: src/plugins/rename/testmod_rename.c [hunk 10]
```diff
@@ -395,6 +395,26 @@ static void test_mixCase ()
 	PLUGIN_CLOSE ();
 }
 
+static void test_write ()
```
Comment (markus2330) [inline src/plugins/rename/testmod_rename.c]: can you also test the other direction (upper/lower)

**Comment by tom-wa** [comment]
@markus2330 done

**Comment by markus2330** [comment]
Thank you!
